Refactor/sheet builder implementation (#13)
ronimizy/FluentSpreadsheets: README.md

  ```
## Sheets API

There are two main kinds of sheet segments:

> - A segment that consists of a single data column:
>   - `SegmentBase`
>   - `HeaderRowSegmentBase`

Author: ```suggestion
- A segment that consists of a single data column:
  - `SegmentBase`
  - `RowSegmentBase`
  - `HeaderRowSegmentBase`
```

Author: + list of methods to implement

Reviewer: > + list of methods to implement

You mean right here for every segment?

- A segment that can represent a collection of data columns:
  - `PrototypeSegmentBase`
  - `PrototypeHeaderRowSegmentBase`
  

### SegmentBase

> - Requires you to implement 2 methods

Author: ```suggestion
- Requires you to implement 3 methods
```

    - `IComponent BuildHeader(THeaderData data)`
    - `IComponent BuildRow(HeaderRowData<THeaderData, TRowData> data, int rowIndex)`
    - `IComponent BuildFooter(HeaderFooterData<THeaderData, TFooterData> data)`

### HeaderRowSegmentBase

- Requires you to implement 2 methods
    - `IComponent BuildHeader(THeaderData data)`
    - `IComponent BuildRow(HeaderRowData<THeaderData, TRowData> data, int rowIndex)`

### PrototypeSegmentBase

Prototype sheet segments are useful when your table may have a dynamic amount of columns depending on the data that has
been provided.

Prototype segment must have a collection of header data to create multiple segments, so

> you must implement `IEnumerable<TSourceHeaderData> SelectHeaderData(THeaderData data)` method to extract collection

Author: The method called just `Select`, but it would be better off called `SelectHeaderData`.

of header data that prototypes will use from general header data.

> You also can implement `IPrototypeSegmentHeaderCustomizer<TData>` and customize header, which will be shown over all cells. 

Author: ```suggestion
You also can implement `IPrototypeSegmentHeaderCustomizer<TData>` and customize header component (all header components in an `HStack`), all changes allowed, except ones that will increase header's width.
```

> ### PrototypeSegmentBase

Author: ```suggestion
### PrototypeHeaderRowSegmentBase
```

Works like PrototypeSegmentBase, but there is no footer and you must implement \

> `IEnumerable<TDestinationHeaderData> Select(TSourceHeaderData source)`

Author: I will rename the method to `SelectHeaderData`
